src/modular/bernstein_yang/boxed.rs

```diff
@@ -68,17 +71,14 @@ impl Inverter for BoxedBernsteinYangInverter {
 
 /// Returns the greatest common divisor (GCD) of the two given numbers.
 pub(crate) fn gcd(f: &BoxedUint, g: &BoxedUint) -> BoxedUint {
-    // TODO (#312): depending on the decision there, this may be relaxed.
-    if f.nlimbs() != g.nlimbs() {
-        panic!("Both arguments to `gcd()` must have the same number of limbs");
-    }
+    let nlimbs = bernstein_yang_nlimbs!(max(f.nlimbs(), g.nlimbs()) * Limb::BITS as usize);
 
     let bits_precision = f.bits_precision();
     let inverse = inv_mod2_62(f.as_words());
-    let f = BoxedUnsatInt::from(f);
-    let mut g = BoxedUnsatInt::from(g);
-    let mut d = BoxedUnsatInt::zero(f.nlimbs());
-    let e = BoxedUnsatInt::one(f.nlimbs());
+    let f = BoxedUnsatInt::from(f).widen(nlimbs);
+    let mut g = BoxedUnsatInt::from(g).widen(nlimbs);
```

Comment:
This pattern could probably benefit from a conversion that allocates `nlimbs` in the first place, but that will be easier to add after this change.